hathor/p2p/sync_checkpoints.py

```diff
@@ -0,0 +1,306 @@
+# Copyright 2021 Hathor Labs
```

Comment:
I feel we should add some safeguards when it comes to partially validated transactions:

1) Add a temporary `PARTIALLY_VALIDATED` item to `meta.voided_by`.
2) Maybe add a parameter `allow_partial_validation=False` in `tx_storage.get_transaction(...)` method. If a transaction is partially validated, raise a `TxNotFound` exception. This would force us to allow loading this transaction only on sync-v2 parts, preventing APIs from retrieving partially validated transactions.
3) Add an assert in the APIs preventing them from returning partially validated transactions.
4) Prevent reliable integrations from generating events for partially validated transactions.

Reply:
Those sound great. I'm doing 1 and 2, and I'll try to do the others as much as I can on this PR.

Reply:
1 is out, 2 in progress

Reply:
2 is out. I didn't try to do 3 through 6, even though they are simple, I'll leave this comment thread open while we discuss this, but I want to track this somewhere else so it gets done during the next series of improvements to the sync-v2.

Reply:
Great! These protections were implemented on separated PRs:

1. https://github.com/HathorNetwork/hathor-core/pull/577
2. https://github.com/HathorNetwork/hathor-core/pull/617